feed/script.js

```diff
@@ -0,0 +1,205 @@
+];
+
+tabsData.forEach((tab) => {
+  const listItem = document.createElement('li');
+  listItem.textContent = tab.name;
+  listItem.dataset.type = tab['data-type'];
+  if (tab.class) {
+    listItem.classList.add(tab.class);
+  }
+  tabsList.appendChild(listItem);
+});
+
+const tabs = document.querySelectorAll('.tabs li');
+
+tabs.forEach((tab) => {
+  tab.addEventListener('click', async function () {
+    tabs.forEach((t) => t.classList.remove('active'));
+    this.classList.add('active');
+    category = this.dataset.type;
+    changeFilter();
+    populateActivityFeed({ category });
+  });
+});
+
+const changeFilter = () => {
+  nextLink = '';
+  activityFeedContainer.innerHTML = '';
+};
+
+async function renderFeed() {
+  changeFilter();
+  populateActivityFeed({ category });
+  addIntersectionObserver();
+}
+
+const intersectionObserver = new IntersectionObserver(async (entries) => {
+  if (!nextLink) {
+    return;
+  }
+  if (entries[0].isIntersecting && !isDataLoading) {
+    await populateActivityFeed({ category }, nextLink);
+  }
+});
+
+const addIntersectionObserver = () => {
+  intersectionObserver.observe(lastElementContainer);
+};
+
+const removeIntersectionObserver = () => {
+  intersectionObserver.unobserve(lastElementContainer);
+};
+
+// main entry
+renderFeed();
+
+function renderActivityItem(data) {
+  const item = document.createElement('li');
+  item.classList.add('activity-item');
+  item.innerHTML = '';
+  let content = '';
+  switch (data.type) {
+    case logType.EXTENSION_REQUESTS:
+      content = formatExtensionRequestsLog(data);
+      break;
+    case logType.TASK:
+      content = formatTasksLog(data);
+      break;
+    case logType.REQUEST_CREATED:
+    case logType.REQUEST_APPROVED:
+    case logType.REQUEST_REJECTED:
+      content = formatOOORequests(data);
+      break;
+    default:
+      content = 'Unknown activity type';
+  }
+  item.innerHTML = `
+    ${content}`;
+  return item;
+}
+
+function formatOOORequests(data) {
+  let text = '';
+  if (data.type === logType.REQUEST_CREATED)
+    text = `<p><strong>${data?.user || data?.createdBy
+      }</strong> raised an OOO request from <strong>${new Date(
+        data.from,
+      ).toLocaleString()}</strong> until <strong>${new Date(
+        data.until,
+      ).toLocaleString()}</strong></p>`;
+  if (data.status === 'APPROVED')
+    text = `${data?.user || data?.createdBy
+      } approved an extension request for task:`;
+  if (data.status === 'DENIED')
+    text = `${data?.user || data?.createdBy
+      } rejected an extension request for task:`;
+
+  return `<div>
+            <div class="title">
+              <img src="assets/leave.png" class="img_icon">
+              <p class="">${text}</p>
+           </div>
+           ${data.message ? `<strong>Reason:</strong> ${data.message}` : ''}
+           <p class="timestamp">
+          ${new Date(data.timestamp * 1000)}
+          </p>
+       </div>`;
+}
+
+function formatExtensionRequestsLog(data) {
```

Comment:
you might consider adding checks for unexpected data.status values to handle potential edge cases more gracefully

Reply:
Noted.